opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/jdbc/JdbcAspect.java

```diff
@@ -0,0 +1,38 @@
+package io.opentracing.contrib.spring.cloud.jdbc;
+
+import io.opentracing.contrib.jdbc.TracingConnection;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+
+/**
+ * Spring AOP Aspect wrapping JDBC-related calls, delegating as much as possible to the official OpenTracing Java JDBC
+ * framework integration.
+ *
+ * @author Juraci Paixão Kröhling
+ */
+@Aspect
+public class JdbcAspect {
+
+  /**
+   * Intercepts calls to {@link DataSource#getConnection()} (and related), wrapping the outcome in a {@link TracingConnection}
+   * @param pjp the intercepted join point
+   * @return  a new {@link TracingConnection} wrapping the result of the joint point
+   */
+  @Around("execution(java.sql.Connection *.getConnection(..)) && target(javax.sql.DataSource)")
+  public Object getConnection(final ProceedingJoinPoint pjp) throws Throwable {
+    Connection conn = (Connection) pjp.proceed();
+    String url = conn.getMetaData().getURL();
+    String user = conn.getMetaData().getUserName();
+    String dbType;
+    try {
+      dbType = url.split(":")[1];
```

Comment:
possible index out of bound exception?

Reply:
That's the reason it's wrapped in a speciall try/catch instead of just letting the exception blow. This is consistent with the OT-JDBC, which would also fail on this situation.

Reply:
isn't it better from the performance perspective do if checks rather than throwing an exception? hmm but I guess this would fail only once or so. 

hmm2: I think it's better to do checks and log if the url is wrong. If the url is wrong just return the connection. It should fail anyway no? But in different place

Reply:
If the same change is applied to the JDBC one, I'm fine with that. But I won't make this behave differently than that one.

Reply:
> If the same change is applied to the JDBC one, I'm fine with that. But I won't make this behave differently than that one.

Why? if it will fail anyway I think it's better to wait for  exception from JDBC/spring.

Reply:
https://github.com/opentracing-contrib/java-jdbc/blob/master/src/main/java/io/opentracing/contrib/jdbc/TracingDriver.java#L116 -- This is what I mean :)

Reply:
I know I am just asking why? 
.... 
>  if it will fail anyway I think it's better to wait for exception from JDBC/spring.